documentation/projects/proposals/relicensing/20241028-implementation_plan_relicensing.md

```diff
@@ -0,0 +1,322 @@
+be filled out):
+
+```md
+The default license for this project is
+[GPL-3.0-or-later](https://spdx.org/licenses/GPL-3.0-or-later.html).
+
+The following libraries are licensed under
+[LGPL-3.0-or-later](https://spdx.org/licenses/LGPL-3.0-or-later.html) and not
+under the default license.
+
+- <libraries' names go here>
+
+The following applications are licensed under
+[AGPL-3.0-or-later](https://spdx.org/licenses/AGPL-3.0-or-later.html) and not
+under the default license.
+
+- <applications' names go here>
+
+## History
+
+All code before <commit> is available under
+[MIT](https://spdx.org/licenses/MIT.html).
+```
+
+#### Add/update `LICENSE` files
+
+We will update the existing root license and add the appropriate licenses for
+the code blocks that were exempted from the default and applied specific
+licenses.
+
+- `api/LICENSE` (new, AGPL-3.0-or-later)
+- `frontend/LICENSE` (new, AGPL-3.0-or-later)
+- `packages/js/api-client/LICENSE` (new, LGPL-3.0-or-later)
+- `packages/python/openverse-attribution/LICENSE` (new, LGPL-3.0-or-later)
+- `LICENSE` (MIT → GPL-3.0-or-later)
+
+#### Update license in manifests
+
+The following files contain references to the project license. These will need
+to be updated.
+
+- Python packages
+  - `.vale/pyproject.toml`
+  - `api/pyproject.toml`
+  - `documentation/pyproject.toml`
+  - `indexer_worker/pyproject.toml`
+  - `packages/python/openverse-attribution/pyproject.toml`
+- JS packages
+  - `packages/js/api-client/package.json`
+  - `packages/js/k6/package.json`
+- Other references
+  - `documentation/packages/js/api_client/index.md`
+  - `api/conf/settings/spectacular.py`
+  - `catalog/tests/factories/github/pull.json`
+
+## Communications
+
+<!-- Note any projects this plan depends on. -->
+
+Both at the start of the migration, and once the migration is completed, we will
+need to communicate the status to the
+[stakeholders listed in the project proposal](/projects/proposals/relicensing/20241028-project_proposal.md#participants-and-stakeholders).
+
+The following communication will be needed
+
+- intent to relicense
+
+  This should be communicated once this project proposal and implementation plan
+  are approved and merged.
+
+  This notification should clearly state
+
+  - the new GNU licenses that we are planning to switch to for the various code
+    blocks
+  - that the change only applies to future commits after the license change
+  - that code before the license change can continue to be used under the same
+    terms
+
+  This also gives us time to see the community's reception to the changes.
+
+  I do not anticipate any negative reactions because, unlike the license changes
+  at Elasticseach, Redis and others, we are not making the project closed-source
+  but rather making it more open-source
+  ([in a sense](https://drewdevault.com/2024/04/19/2024-04-19-copyleft-is-not-restrictive.html))
+  by adding stronger copyleft protections.
```

Comment:
We could add a clause like: "If concerns are raised during the communication phase, we will convene a stakeholder discussion to address and potentially adjust the relicensing strategy." to make sure that we account for any pushback (I also don't anticipate any, but still).

Reply:
Added in afac0ee.